docs/extensions.md

```diff
@@ -0,0 +1,365 @@
+# New Relic CLI Extensions
+
+## Overview
+The New Relic CLI provides an extension framework as a way to integrate new commands into its core command set.  Extensions can be built in any language.
+
+## Getting started
+
+This section will guide you through creating an extension for the New Relic CLI.  At its core, an extension is a repository that contains an `extension.yml` in its root, and a way to invoke a process that will start the extension.
+
+### Plugin modes
+There are two execution models for CLI extensions, API mode and binary execution mode.
+
+#### API mode
+In this execution model, the extension is responsible for communicating with the CLI host via an HTTP API.  The API allows for rich interaction with the user (e.g. interactive prompts), conduits for displaying log messages, warnings, errors, and command output, and a means of communicating to the New Relic backend via Nerdgraph queries.
+
+The interaction begins with a call to [`/command`](#command), which will return the command that the user has invoked along with any provided flags.  The extension can then carry out its work and pass its output to the user via the [`/out`](#out) endpoint.  See the [CLI API reference](#cli-api) section below for more details on communicating with the CLI API.
+
+#### Binary execution mode
+In this execution model, the extension simply receives its command and arguments as command line arguments during invocation.  IO from the extension is redirected to the user's terminal.
+
+When using `binary` mode, the extension's entrypoint (see [entrypoint](#entrypoint) above) will be invoked with additional arguments representing the command to be executed and a collextxion of flag value pairs, in the format `<ENTRYPOINT> <COMMAND> <FLAG_1> <VALUE_1> <FLAG_2> <VALUE_2> <FLAG_N> <VALUE_N>`.  An example invocation:
```

Comment:
```diff
- collextxion 
+ collection
```

Reply:
Not a big deal here since this is just to convey a general idea, but we might want to include the technical term of "argument" in the signature format example above.

e.g.
```
<ENTRYPOINT> <COMMAND> <ARGUMENT> <FLAG_1> <VALUE_1> <FLAG_2> <VALUE_2> <FLAG_N> <VALUE_N>
```